src/brucke_filter.erl

```diff
@@ -0,0 +1,76 @@
+%%%
+%%%   Copyright (c) 2016-2017 Klarna AB
+%%%
+%%%   Licensed under the Apache License, Version 2.0 (the "License");
+%%%   you may not use this file except in compliance with the License.
+%%%   You may obtain a copy of the License at
+%%%
+%%%       http://www.apache.org/licenses/LICENSE-2.0
+%%%
+%%%   Unless required by applicable law or agreed to in writing, software
+%%%   distributed under the License is distributed on an "AS IS" BASIS,
+%%%   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%%%   See the License for the specific language governing permissions and
+%%%   limitations under the License.
+%%%
+
+-module(brucke_filter).
+
+-export([ init/2
+        , init/3
+        , filter/5
+        , filter/6
+        ]).
+
+-export_type([ filter_result/0
+             ]).
+
+-include("brucke_int.hrl").
+-include_lib("brod/include/brod.hrl").
+
+-type filter_result() :: boolean() | {kafka_key(), kafka_value()}.
+
+%% Called when route worker (`brucke_member') start/restart.
+%% The assumption now is that the callback implementor should be stateless.
+%% This is why is no support for callback state returned. i.e. the filter
+%% is designed to be a `filter' not intended to be a `fold'.
+%% In case the callback implementor needs to be stateful, they will have
+%% to register global names etc.
```

Comment:
`brucke_filter` works like a `lists:filtermap`, not `lists:foldl` --- stateless.
A more generic solution would be to allow `CallbackState` returned from `init/2` and and mutated in `filter/5`, but looks like an overkill and unnecessary complication.
This is why if the users need to be stateful, they can choose global names for process or ets etc.